src/common/pf_lldp.c

```diff
@@ -1020,6 +1020,108 @@ bool pf_lldp_is_peer_info_received (pnet_t * net, uint32_t * timestamp_10ms)
    return is_received;
 }
 
+static void pf_lldp_send_remote_mismatch_alarm (
+   pnet_t * net,
+   const pnet_lldp_peer_info_t * lldp_peer_info)
+{
+   uint16_t ix = 0;
+   pf_ar_t * p_ar = NULL;
+   bool peer_stationname_is_correct = false;
+   bool peer_portname_is_correct = false;
+   bool found_active_ar = false;
+   pnet_diag_source_t diag_source = {0};
+   pf_check_peer_t * p_wanted_peer = &net->port[0].check.peer;
+
+   if (net->port[0].check.active == true)
+   {
+      if (
+         strncmp (
+            lldp_peer_info->chassis_id.string,
+            (char *)p_wanted_peer->peer_station_name,
+            lldp_peer_info->chassis_id.len) == 0)
+      {
+         peer_stationname_is_correct = true;
+      }
+
+      // TODO Update also peer_portname_is_correct
+
+      for (ix = 0; ix < NELEMENTS (net->cmrpc_ar); ix++)
```

Comment:
The observer checking an alarm rule should not need to care whether there is an active ar or not. I think check should be inside alarm, or the loop will be common to all observers and also application/process alarms?

On the other hand what will happen if we do not have an active ar when the check is performed and the lldp frame is received - then we will never send any alarm ( If I do not misunderstand).